zerver/models/sessions.py

```diff
@@ -0,0 +1,122 @@
+# https://github.com/typeddjango/django-stubs/issues/1698
+# mypy: disable-error-code="explicit-override"
+
+from django.conf import settings
+from django.contrib.sessions.backends.cached_db import SessionStore as CachedDbSessionStore
+from django.contrib.sessions.base_session import AbstractBaseSession
+from django.db import models
+from django.db.models.signals import post_delete, pre_save
+from django.db.transaction import get_connection
+from django.dispatch import receiver
+from typing_extensions import override
+
+from zerver.models.realms import Realm
+from zerver.models.users import UserProfile
+
+
+# custom session model
+class RealmSession(AbstractBaseSession):
+    realm = models.ForeignKey(Realm, null=True, on_delete=models.CASCADE)
+    user = models.ForeignKey(UserProfile, null=True, on_delete=models.CASCADE)
+    ip_address = models.GenericIPAddressField(null=True)
+
+    class Meta:
+        indexes = [
+            models.Index(fields=("realm", "user"), name="zerver_realmsession_realm_user_idx"),
+            models.Index(fields=("ip_address",), name="zerver_realmsession_ip_address_idx"),
```

Comment:
This one seems equivalent to setting `db_index=True` on the `ip_address` field?

Reply:
Yes, when we want to just apply a single simple index with no  customization, `db_index=True` is more convenient